development/tools/pkg/release/storage.go

```diff
@@ -0,0 +1,49 @@
+package release
+
+import (
+	"context"
+	"io/ioutil"
+
+	"cloud.google.com/go/storage"
+	"github.com/kyma-project/test-infra/development/tools/pkg/common"
+)
+
+// StorageAPI exposes a function to read objects from Google Storage buckets
+type StorageAPI interface {
+	ReadBucketObject(ctx context.Context, fileName string) ([]byte, error)
+}
+
+// storageAPIWrapper implements reading objects from Google Storage buckets
+type storageAPIWrapper struct {
+	storageClient *storage.Client
+	bucketName    string
+	folderName    string
+}
+
+// NewStorageAPI returns implementation of storageAPI
+func NewStorageAPI(ctx context.Context, bucketName, folderName string) (StorageAPI, error) {
+
+	storageClient, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &storageAPIWrapper{
+		storageClient: storageClient,
+		bucketName:    bucketName,
+		folderName:    folderName,
+	}, nil
+}
+
+// ReadBucketObject downloads and saves in temporary directory a file specified by fileName from a given bucket
+func (saw *storageAPIWrapper) ReadBucketObject(ctx context.Context, fileName string) ([]byte, error) {
```

Comment:
It would be awesome if `ReadBucketObject` could return `io.Reader` instead of `bytes[]`. That could simplify passing artifacts to GitHub. I see two ways for improvement:
1. Instead of passing bytes around you stream directly from bucket to tmp file
2. You can go even further and stream directly to github, skipping temp files as mentioned [here](https://github.com/google/go-github/blob/master/github/repos_releases.go#L340), but that would require you to know size of the file upfront.